iroh-store/src/config.rs

```diff
@@ -17,6 +17,14 @@ pub const CONFIG_FILE_NAME: &str = "store.config.toml";
 /// For example, `IROH_STORE_PATH=/path/to/config` would set the value of the `Config.path` field
 pub const ENV_PREFIX: &str = "IROH_STORE";
 
+/// the path to data directory. If arg_path is `None`, the default iroh_data_path()/store is used
+/// iroh_data_path() returns an operating system-specific directory
+/// falls back to an empty path in the unlikely scenario where iroh_data_path() is `None`
+/// likely b/c iroh is not running on a Linux/Windows/MacOS system
+pub fn config_data_path(arg_path: Option<PathBuf>) -> PathBuf {
+    arg_path.unwrap_or_else(|| iroh_data_path("store").unwrap_or_else(|| PathBuf::from("")))
+}
```

Comment:
unclear on if this should live here, or exist at all. the nested `unwrap_or_else` also feels smelly

Reply:
we should error out and not fall back to an empty path, that is very surprising

Reply:
It errors out lower in the stack because it cant use an "" path, but yeah should be handled higher up.

Reply:
ok, changes made. latest refactor handles this by erroring out when no path is present. error check happens before DB creation.